neon_utils/signal_utils.py

```diff
@@ -0,0 +1,167 @@
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+from time import time, sleep
+
+from mycroft_bus_client import MessageBusClient, Message
+import ovos_utils.signal
+
+from neon_utils.logger import LOG
+
+BUS = MessageBusClient()
+BUS.run_in_thread()
+
+
+def fs_wait_for_signal_create(signal_name: str, timeout: int = 30):
+    expiration = time() + timeout
+    while not check_for_signal(signal_name, -1) and time() < expiration:
+        sleep(0.1)
+    return check_for_signal(signal_name, -1)
+
+
+def fs_wait_for_signal_clear(signal_name: str, timeout: int = 30):
+    expiration = time() + timeout
+    while check_for_signal(signal_name, -1) and time() < expiration:
+        sleep(0.1)
+    return check_for_signal(signal_name, -1)
+
+
+# Init default file handler methods
+create_signal = ovos_utils.signal.create_signal
+check_for_signal = ovos_utils.signal.check_for_signal
+wait_for_signal_clear = fs_wait_for_signal_clear
+wait_for_signal_create = fs_wait_for_signal_create
+
+
+def init_signal_handlers():
+    """
+    Initialize the proper signal methods dependent on the Signal Manager being available.
+    Any previously imported methods will not be affected by calls to this method, but imports of `signal_utils` are.
+    i.e. `from neon_utils.signal_utils import check_for_signal`, `check_for_signal` is not changed,
+         `import neon_utils.signal_utils`, `neon_utils.signal_utils.check_for_signal` is changed.
+    """
+    global create_signal
+    global check_for_signal
+    global wait_for_signal_clear
+    global wait_for_signal_create
+    if check_signal_manager_available():
+        LOG.info("Signal Manager Available")
+        create_signal = manager_create_signal
+        check_for_signal = manager_check_for_signal
+        wait_for_signal_clear = manager_wait_for_signal_clear
+        wait_for_signal_create = manager_wait_for_signal_create
+    else:
+        LOG.warning("No signal manager available; falling back to FS signals")
```

Comment:
this makes me think that both `fs_XXX` and `manager_XXX` should be private methods

I would like to keep public methods to a minimum here and discourage usage of signals in general. the concept of "activity" that mycroft-core is pushing largely makes this a backwards compat thing. I think almost every single signal has an equivalent bus message already that can be listened for

signals still have their place for usage where an event based approach isn't the best (even if internally it is still event based it doesnt change paradigm, its only a function call)

Reply:
That makes sense; there is no reason (outside of testing) that anyone would use those methods directly.. I'll refactor those to be private.